src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -114,6 +116,34 @@ You can use \`IUiSettingsClient.get("${key}", defaultValue)\`, which will just r
     return this.resolveValue(value, type);
   }
 
+  getWithBrowserSettings<T = any>(key: string, defaultOverride?: T) {
+    const declared = this.isDeclared(key);
+
+    if (!declared && defaultOverride !== undefined) {
+      return defaultOverride;
+    }
+
+    if (!declared) {
+      throw new Error(
+        `Unexpected \`IUiSettingsClient.get("${key}")\` call on unrecognized configuration setting "${key}".`
+      );
+    }
+
+    const type = this.cache[key].type;
+    const userValue = this.cache[key].userValue;
```

Comment:
i wouldn't name variable userValue as its really the value from advanced settings, not related to the user